src/main/java/com/linkedin/kmf/apps/MirrorPipelineMonitor.java

```diff
@@ -0,0 +1,448 @@
+    }
+  }
+
+  private Map<String, Object> createProduceServiceConfig(Map<String, Object> props) {
+    Map<String, Object> produceServiceConfig = getProps(props, MirrorPipelineMonitorConfig.PRODUCE_SERVICE_CONFIG);
+
+    if (produceServiceConfig == null) {
+      throw new RuntimeException(_name + ": Must specify a produce service config.");
+    }
+
+    produceServiceConfig.put(CommonServiceConfig.TOPIC_CONFIG, props.get(CommonServiceConfig.TOPIC_CONFIG));
+
+    return produceServiceConfig;
+  }
+
+  private Map<String, Object> createConsumeServiceConfig(Map<String, Object> props) {
+    Map<String, Object> consumeConfig = getProps(props, MirrorPipelineMonitorConfig.CONSUME_SERVICE_CONFIG);
+
+    if (consumeConfig == null) {
+      throw new RuntimeException(_name + ": Must specify a consume service config.");
+    }
+
+    consumeConfig.put(CommonServiceConfig.TOPIC_CONFIG, props.get(CommonServiceConfig.TOPIC_CONFIG));
+
+    return consumeConfig;
+  }
+
+  private Map<String, Object> createTopicManagementServiceConfig(Map<String, Object> props) {
+    Map<String, Object> topicMangementConfig = getProps(props, MirrorPipelineMonitorConfig.TOPIC_MANAGEMENT_SERVICE_CONFIG);
+
+    if (topicMangementConfig == null) {
+      throw new RuntimeException(_name + ": Must specify a topic management service config.");
+    }
+
+    topicMangementConfig.put(CommonServiceConfig.TOPIC_CONFIG, props.get(CommonServiceConfig.TOPIC_CONFIG));
+
+    return topicMangementConfig;
+  }
+
+  @Override
+  public void start() {
+    _produceService.start();
+    _consumeService.start();
+    for (TopicManagementService topicManagementService : _topicManagementServices) {
+      topicManagementService.start();
+    }
+    LOG.info(_name + "/MirrorPipelineMonitor started");
+  }
+
+  @Override
+  public void stop() {
+    for (TopicManagementService topicManagementService : _topicManagementServices) {
+      topicManagementService.stop();
+    }
+    _produceService.stop();
+    _consumeService.stop();
+    LOG.info(_name + "/MirrorPipelineMonitor stopped");
+  }
+
+  @Override
+  public boolean isRunning() {
+    for (TopicManagementService topicManagementService : _topicManagementServices) {
+      if (!topicManagementService.isRunning()) {
+        return false;
+      }
+    }
+
+    return _produceService.isRunning() && _consumeService.isRunning();
+  }
+
+  @Override
+  public void awaitShutdown() {
+    for (TopicManagementService topicManagementService : _topicManagementServices) {
+      topicManagementService.awaitShutdown();
+    }
+    _produceService.awaitShutdown();
+    _consumeService.awaitShutdown();
+  }
+
+  @SuppressWarnings("unchecked")
+  private Map<String, Object> getProps(Map<String, Object> props, String key) {
+    if (props.containsKey(key)) {
+      return (Map<String, Object>) props.get(key);
+    }
+
+    return null;
+  }
+
+  @SuppressWarnings("unchecked")
+  private List<String> getList(Map<String, Object> props, String key) {
```

Comment:
You can probably used `getList()` defined in `MirrorPipelineMonitorConfig` after `MirrorPipelineMonitorConfig` extends `AbstractConfig`. See `TopicManagementService` for example usage. Same suggestion for `getDouble()`.